src/components/profile/AskerRegistration.tsx

```diff
@@ -33,28 +33,28 @@ import { AgencySelection } from '../agencySelection/AgencySelection';
 import './profile.styles';
 import { apiGetUserData } from '../../api';
 import { Text, LABEL_TYPES } from '../text/Text';
-import { isGroupChatConsultingType } from '../../utils/resorts';
 import { Headline } from '../headline/Headline';
-import { apiGetConsultingType } from '../../api/apiGetConsultingType';
-import { ConsultingTypeInterface } from '../../globalState';
+import {
+	ConsultingTypesContext,
+	useConsultingType
+} from '../../globalState/provider/ConsultingTypesProvider';
 
 export const AskerRegistration = () => {
 	const { userData, setUserData } = useContext(UserDataContext);
 	const [isButtonDisabled, setIsButtonDisabled] = useState(true);
-	const [selectedConsultingType, setSelectedConsultingType] = useState<
+	const [selectedConsultingTypeId, setSelectedConsultingTypeId] = useState<
 		number
 	>(null);
-	const [
-		selectedConsultingTypeData,
-		setSelectedConsultingTypeData
-	] = useState<ConsultingTypeInterface>();
 	const [selectedAgency, setSelectedAgency] = useState<any>({});
 	const [overlayActive, setOverlayActive] = useState(false);
 	const [overlayItem, setOverlayItem] = useState<OverlayItem>(null);
 	const { setAcceptedGroupId } = useContext(AcceptedGroupIdContext);
 	const [isRequestInProgress, setIsRequestInProgress] = useState(false);
+	const { consultingTypes } = useContext(ConsultingTypesContext);
+	const selectedConsultingType = useConsultingType(selectedConsultingTypeId);
 
-	const isAllRequiredDataSet = () => selectedConsultingType && selectedAgency;
+	const isAllRequiredDataSet = () =>
+		selectedConsultingTypeId && selectedAgency;
```

Comment:
Should this be:

```suggestion
		selectedConsultingTypeId != null && selectedAgency;
```

? Or is there special behaviour for the consulting type `0`?

Reply:
Nope, you are correct, please commit this change!